test-bitcoincore-rpc/src/lib.rs

```diff
@@ -282,6 +292,52 @@ impl Api for Server {
       }
     }
   }
+
+  fn list_unspent(
+    &self,
+    minconf: Option<usize>,
+    maxconf: Option<usize>,
+    address: Option<bitcoin::Address>,
+    include_unsafe: Option<bool>,
+    query_options: Option<String>,
+  ) -> Result<Vec<ListUnspentResultEntry>, jsonrpc_core::Error> {
+    assert_eq!(minconf, None, "minconf param not supported");
+    assert_eq!(maxconf, None, "maxconf param not supported");
+    assert_eq!(address, None, "address param not supported");
+    assert_eq!(include_unsafe, None, "include_unsafe param not supported");
+    assert_eq!(query_options, None, "query_options param not supported");
+    let all_outpoints = self
+      .state
+      .lock()
+      .unwrap()
+      .transactions
+      .iter()
+      .flat_map(|(txid, tx)| {
+        (0..tx.output.len()).map(|vout| bitcoin::OutPoint::new(*txid, vout as u32))
+      })
+      .collect::<Vec<OutPoint>>();
+
+    Ok(
+      all_outpoints
+        .iter()
+        .map(|outpoint| ListUnspentResultEntry {
+          txid: outpoint.txid,
+          vout: outpoint.vout,
+          address: None,
+          label: None,
+          redeem_script: None,
+          witness_script: None,
+          script_pub_key: Script::new(),
+          amount: Amount::default(),
+          confirmations: 0,
+          spendable: true,
+          solvable: true,
+          descriptor: None,
+          safe: true,
+        })
+        .collect(),
+    )
```

Comment:
Not sure if this is right:

```suggestion
    Ok(
      self
        .state
        .lock()
        .unwrap()
        .transactions
        .iter()
        .flat_map(|(txid, tx)| {
          (0..tx.output.len()).map(|vout| ListUnspentResultEntry {
          txid: outpoint.txid,
          vout: outpoint.vout,
          address: None,
          label: None,
          redeem_script: None,
          witness_script: None,
          script_pub_key: Script::new(),
          amount: Amount::default(),
          confirmations: 0,
          spendable: true,
          solvable: true,
          descriptor: None,
          safe: true,
        })
      })
      .collect()
    )
```